src/OrchardCore.Modules/OrchardCore.Secrets/Stores/DatabaseSecretStore.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.Extensions.Localization;
+using Newtonsoft.Json;
+using OrchardCore.Secrets.Models;
+using OrchardCore.Secrets.Services;
+
+namespace OrchardCore.Secrets.Stores;
+
+public class DatabaseSecretStore : ISecretStore
+{
+    private readonly SecretsDocumentManager _manager;
+    private readonly IDataProtector _protector;
+    protected readonly IStringLocalizer S;
+
+    public DatabaseSecretStore(
+        SecretsDocumentManager manager,
+        IDataProtectionProvider dataProtectionProvider,
```

Comment:
Why is `IDataProtectionProvider` used? I read it somewhere, but can't find any information about it either way, that it shouldn't be used to persist secrets (just encrypt them on the fly, like within a request or user session).

Reply:
This is a first level of protection as we use to store password, for example a RSA secret may contain a private key.

Then yes, this is when we exchange secrets between instances that we use a kind of envelope containing the data encrypted with a symetric key, but to be exchanged with the same envelope this symetric key is encrypted itself with a dedicated RSA secret, and we also use another RSA secret for signing. Both RSA secrets shared their public key on both sides, the private key being only in one side.

Reply:
According to https://learn.microsoft.com/en-us/aspnet/core/security/data-protection/configuration/overview?view=aspnetcore-7.0#protectkeyswithazurekeyvault persisting these keys is officially supported, thus `IDataProtectionProvider` should be fine.